### ISSUE TITLE
Joint Loss 에 대한 질문이 있습니다

### ISSUE THREAD

**Issue by jsrimr** [ozmig77/dcnet#2]
여기에 이런걸 올리는게 적절한지는 모르겠으나... JointLoss 를 어떻게 생각해내셨는지 궁금합니다. 3.3 의 설명만으로는 잘 이해가 안가서요 ㅠ

단순하게 생각하면, joint_loss 가 뭘 해내줄지 모르겠는데 실험결과를 좋게한다고 하니 정말 대단한것 같습니다.
제가 joint_loss 가 뭘 해내줄지 모르겠는 배경은 아래와 같습니다.
Corretion Network 가 잘 훈련되었다면 t+x_ref 나 gamma*d+(1-gamma)*t + x_ref 가 별 차이가 없을 거기 때문에 C_e 와 C_e* 가 사실상 같은 값입니다. 따라서 Correction Network 가 훈련이 잘 되었을 경우에는 얻는게 없고 Correction Network 가 아직 훈련이 좀 덜 됐을 때 어떤 regularizing 효과를 기대하고 joint_loss 가 의미가 있다는거 같은데 어떤 regularizing effect 를 기대하는건지를 모르겠습니다...

우선, C_e 가 C_e* 에 가까워지는 걸 원하시는건지 그 반대인지도 모르겠구
그림에 나타난 2개의 CompositionNetwork 는 같은 네트워크인지 아니면 주기적으로  weight 를 sync 하는 2개의 다른 네트워크인지도 잘 모르겠고 해서요
```
score_matrix_joint = self.model.get_score_matrix(
                        trg_embds=aux_trg, # {mod: B x N x D or N x D}
                        src_embds=aux_src, # {mod: B x D} 
                        subspaces=self.model.modalities,
                    )
                    joint_loss = self.loss(score_matrix_joint)
```
를 보면 한개의 동일한 네트워크 같지만

아무튼 어떻게 joint_loss 를 생각해내신건지, joint_loss 를 통해 뭘 얻고자 하는건지 그 함의에 대해 좀 더 쉬운설명..? 을 얻을 수 있을까요

**Comment by ozmig77** [comment]
먼저 clarify하자면, joint loss는 C_e 가 C_e*를 가깝게 하도록 의도합니다.
그리고 그림에 나타난 2개의 composition network는 동일한 parameter를 사용했습니다.

해당 loss는 joint training시 두 network가 잘 align 되는것을 의도하였습니다.
아이디어는 cycle consistency에서 착안했습니다. 기존 cycle consistency는 correction network의 결과로 나온 output을 다시 composition network에 넣고 그 결과를 기존 image와 비슷하게 만듭니다. 안정성을 위해 joint loss는 기존 image대신 composition network의 output과 비슷하게 만들도록 했습니다.

joint loss는 correction network와 composition network를 모두 사용하여 얻은 output에 기반하는 것을 알 수 있습니다. 
Correction Network 측면에서 살펴보면,
말씀해주신것처럼, Correction Network 가 잘 훈련되었다면 t+x_ref 나 gamma*d+(1-gamma)t + x_ref 가 별 차이가 없을 거기 때문에 C_e* 와 C_e 가 사실상 같은 값입니다. 달리 말하면, C_e*와 C_e가 같아지게 만드는 loss는 Correction Network의 output d가 t와 같아지도록 해준다는 것을 알 수 있습니다. 기존 corr loss와는 다르게 composition net의 output을 통해 loss가 계산되기 때문에 joint training시  network들이 더 잘 align되게 도와줍니다.

Composition Network 측면에서  text를 기준으로 보면,
C_e*는 C_e의 text input에 d라는 perturbation이 추가된 경우로 보실 수 있고, 좀 더 text 변화에 robust하도록 만드는 효과를 얻을 수 있습니다.

답변이 잘 되었으면 좋겠네요. 감사합니다.

**Comment by jsrimr** [comment]
친절한 답변 감사합니다!
